File: .tmp_py4013/batch_13_dump/3460_longest_common_prefix_after_at_most_one_removal.js
// LeetCode 3460 - Longest Common Prefix After at Most One Removal
// https://leetcode.com/problems/longest-common-prefix-after-at-most-one-removal/

var longestCommonPrefix = function(s, t) {
    let i = 0, j = 0;
    let removed = false;
    while (i < s.length && j < t.length) {
        if (s[i] === t[j]) {
            i++;
            j++;
            continue;
        }
        if (removed) break;
        removed = true;
        i++;
    }
    return j;
};

File: .tmp_rs4013/rust_check17/3460_longest_common_prefix_after_at_most_one_removal.rs
struct Solution;
// LeetCode 3460 - Longest Common Prefix After at Most One Removal
// https://leetcode.com/problems/longest-common-prefix-after-at-most-one-removal/

impl Solution {
    pub fn longest_common_prefix(s: String, t: String) -> i32 {
        let sb = s.as_bytes();
        let tb = t.as_bytes();
        let mut i = 0;
        let mut j = 0;
        let mut removed = false;
        while i < sb.len() && j < tb.len() {
            if sb[i] == tb[j] {
                i += 1;
                j += 1;
                continue;
            }
            if removed {
                break;
            }
            removed = true;
            i += 1;
        }
        j as i32
    }
}

fn main() {}
